framework/src/utils/MatrixTools.C

```diff
@@ -31,7 +31,7 @@ int inverse(const std::vector<std::vector<Real> > & m, std::vector<std::vector<R
     for (const auto & matrix_entry : rowvec)
       A.push_back(matrix_entry);
 
-  int error = inverse(A, n);
+  int error = inverse(A, (int) n);
```

Comment:
My suggestion would be to cast as late as possible. We can at least make the interfaces we control look right. Is there a reason for the LAPACK function to accept _signed_ values? Will they return negative values to indicate some error state? If so then we need to trap it and throw an exception (or deal with it in an appropriate way).